spec/system/reports/product_drives_summary_spec.rb

```diff
@@ -0,0 +1,218 @@
+RSpec.describe "Product Drives Summary Report", type: :system, js: true do
+  let!(:storage_location) { create(:storage_location, :with_items, item_quantity: 0, organization: @organization) }
+  let(:org_short_name) { @organization.short_name }
+  let(:org_reports_product_drives_page) { OrganizationReportsProductDrivesSummaryPage.new org_short_name: org_short_name }
+
+  before do
+    @url_prefix = "/#{@organization.short_name}"
+  end
+
+  context "When signed in as a normal user" do
+    before do
+      sign_in @user
+    end
+
+    test_time = Time.zone.now
+
+    # 1, 20, 300, ..., 900000000
+    # assuming each value is used once, summing these values makes easily recognizable totals
+    # .fetch() from it so too-high indices raise IndexError
+    # legal indices are in -8..8 (i.e., inclusive)
+    item_quantities = (1..9).map { |i| i * 10**(i - 1) }
+
+    describe "Product Drives" do
+      around do |example|
+        travel_to(test_time)
+        example.run
+        travel_back
+      end
+
+      it "has a widget for product drive summary data" do
+        org_reports_product_drives_page.visit
+
+        expect(org_reports_product_drives_page).to have_product_drives_section
+      end
+
+      # as of 28 Jan 2022, the "Recent Donations" list shows up to this many items matching the date filter
+      max_recent_donation_links_count = 3
+
+      # Make up to this many (inclusive) donations for each filtered period
+      # Keep it below (item_quantities.size - 1) so there's at least 2 values left for
+      # Donations outside of the filtered period
+      max_donations_in_filtered_period = max_recent_donation_links_count + 1
+
+      def create_next_product_drive_donation(donation_date:)
+        quantity_in_donation = @item_quantity.next
+        drive = @product_drives.sample
+
+        create :product_drive_donation, :with_items, product_drive: drive.drive, product_drive_participant: @product_drive_participant, issued_at: donation_date, item_quantity: quantity_in_donation, storage_location: storage_location, organization: @organization,
+          money_raised: @money_raised_on_each_product_drive
+
+        OpenStruct.new drive_name: drive.name, quantity: quantity_in_donation, money_raised: @money_raised_on_each_product_drive
+      end
+
+      [
```

Comment:
This code seems to be copied multiple times... is there a test method we can extract here?